### PR TITLE
Update issue and PR templates - Close #3302

### LINKED ISSUE #3302: Improve issue templates
### Expected behavior

We can have multiple templates to differentiate issues coming from community with having the definitions below on top of the md files.

```
  ---
title: 'Bug report'
labels: bug
assignees: octocat
  ---
```

See: https://help.github.com/en/articles/manually-creating-a-single-issue-template-for-your-repository

Example: https://github.com/facebook/jest/issues/new/choose 

image

### Actual behavior
We have a single template at the moment.

### PR DESCRIPTION
**Pull request by shuse2**
### What was the problem?
Default issue template was only for Bug report

### How did I fix it?
Add feature requests template
Also, update the PR template to be more specific

### Review checklist

* The PR resolves #3302 
* All new code is covered with unit tests
* All new code was formatted with Prettier
* Linting passes
* Tests pass
* Commit messages follow the [commit guidelines](CONTRIBUTING.md#git-commit-messages)
* Documentation has been added/updated

### CODE REVIEW

File: docs/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -1,15 +1,15 @@
...
 ### Review checklist
 
-* The PR resolves #INSERT_ISSUE_NUMBER
-* All new code is covered with unit tests
-* All new code was formatted with Prettier
-* Linting passes
-* Tests pass
-* Commit messages follow the [commit guidelines](CONTRIBUTING.md#git-commit-messages)
-* Documentation has been added/updated
+- The PR resolves #INSERT_ISSUE_NUMBER
```
Comment (yatki) [inline docs/PULL_REQUEST_TEMPLATE.md]: Converting these item marks to checkbox items can increase awareness of the item: `[ ] -`

File: docs/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -1,15 +1,15 @@
...
+- All new code is covered with unit tests
```
Comment (yatki) [inline docs/PULL_REQUEST_TEMPLATE.md]: Instead we can say code coverage is above 90% maybe ?

File: docs/PULL_REQUEST_TEMPLATE.md [hunk 3]
```diff
@@ -1,15 +1,15 @@
...
+- All new code was formatted with Prettier
+- Linting passes
+- Tests pass
```
Comment (yatki) [inline docs/PULL_REQUEST_TEMPLATE.md]: no need to 3 items above. Git hooks & CI handles them anyway. Just prettier is not executed on ci I think but we can add it.

File: docs/PULL_REQUEST_TEMPLATE.md [hunk 4]
```diff
@@ -1,15 +1,15 @@
...
+- Commit messages follow the [commit guidelines](CONTRIBUTING.md#git-commit-messages)
+- Documentation has been added/updated
```
Comment (yatki) [inline docs/PULL_REQUEST_TEMPLATE.md]: Updating documentation in PR is not a common case. Maybe we should have separate issues labeled as docs instead? Also maybe "Updated JSDoc comments" is a better requirement for us?

Comment (yatki) [reply docs/PULL_REQUEST_TEMPLATE.md]: Some additional rules:

```
[ ] - Synced with target branch
[ ] - Added to relevant Project
[ ] - Added to relevant Milestone
[ ] - Connected to issue on Zenhub
```

File: docs/PULL_REQUEST_TEMPLATE.md (hunk 2 above)
Comment (yatki) [reply docs/PULL_REQUEST_TEMPLATE.md]: Also we can make sure of that by adding as a requirement to merge the PR.

File: docs/PULL_REQUEST_TEMPLATE.md (hunk 4 above)
Comment (yatki) [reply docs/PULL_REQUEST_TEMPLATE.md]:
```
[ ] - "pending review" label was assigned
```

File: docs/PULL_REQUEST_TEMPLATE.md (hunk 2 above)
Comment (shuse2) [reply docs/PULL_REQUEST_TEMPLATE.md]: In total coverage, we don't have 90%

File: docs/PULL_REQUEST_TEMPLATE.md (hunk 4 above)
Comment (shuse2) [reply docs/PULL_REQUEST_TEMPLATE.md]: this is general template, so i don't think adding those will be good.
- JSDocs doesn't apply to elements and commander
- target branch is enforced by the github setting
- relevant project, milestone, zenhub doesn't apply to general contributors

File: docs/ISSUE_TEMPLATE.md [hunk 5]
```diff
@@ -1,7 +1,17 @@
...
+Please describe what functionality is needed
 
-### Which version(s) does this affect? (Environment, OS, etc...)
+### Motivation
+
+Please describe why it is needed
+
+### Acceptance Criteria
+
+Please describe in what conditions this issue will be closed
```
Comment (mitsujutsu) [inline docs/ISSUE_TEMPLATE.md]: Maybe `Please describe the conditions which must be met for this issue to close`, but both are fine

File: docs/PULL_REQUEST_TEMPLATE.md [hunk 6]
```diff
@@ -1,15 +1,13 @@
 ### What was the problem?
 
-### How did I fix it?
+### How did I solve?
```
Comment (mitsujutsu) [inline docs/PULL_REQUEST_TEMPLATE.md]: `How did I solve it?`

**Pull request merged**